public class org.xmlpull.v1.builder.xpath.jaxen.FunctionCallException extends org.xmlpull.v1.builder.xpath.jaxen.JaxenException
{
    private java.lang.Throwable nestedException;

    public void <init>(java.lang.String)
    {
        java.lang.String r1;
        org.xmlpull.v1.builder.xpath.jaxen.FunctionCallException r0;

        r0 := @this: org.xmlpull.v1.builder.xpath.jaxen.FunctionCallException;

        r1 := @parameter0: java.lang.String;

        specialinvoke r0.<org.xmlpull.v1.builder.xpath.jaxen.JaxenException: void <init>(java.lang.String)>(r1);

        return;
    }

    public void <init>(java.lang.Throwable)
    {
        java.lang.Throwable r1;
        java.lang.String $r2;
        org.xmlpull.v1.builder.xpath.jaxen.FunctionCallException r0;

        r0 := @this: org.xmlpull.v1.builder.xpath.jaxen.FunctionCallException;

        r1 := @parameter0: java.lang.Throwable;

        $r2 = virtualinvoke r1.<java.lang.Throwable: java.lang.String getMessage()>();

        specialinvoke r0.<org.xmlpull.v1.builder.xpath.jaxen.JaxenException: void <init>(java.lang.String)>($r2);

        r0.<org.xmlpull.v1.builder.xpath.jaxen.FunctionCallException: java.lang.Throwable nestedException> = r1;

        return;
    }

    public void <init>(java.lang.String, java.lang.Exception)
    {
        java.lang.String r1;
        org.xmlpull.v1.builder.xpath.jaxen.FunctionCallException r0;
        java.lang.Exception r2;

        r0 := @this: org.xmlpull.v1.builder.xpath.jaxen.FunctionCallException;

        r1 := @parameter0: java.lang.String;

        r2 := @parameter1: java.lang.Exception;

        specialinvoke r0.<org.xmlpull.v1.builder.xpath.jaxen.JaxenException: void <init>(java.lang.String)>(r1);

        r0.<org.xmlpull.v1.builder.xpath.jaxen.FunctionCallException: java.lang.Throwable nestedException> = r2;

        return;
    }

    public void printStackTrace(java.io.PrintStream)
    {
        java.lang.Throwable $r1, $r3;
        java.io.PrintStream r2;
        org.xmlpull.v1.builder.xpath.jaxen.FunctionCallException r0;

        r0 := @this: org.xmlpull.v1.builder.xpath.jaxen.FunctionCallException;

        r2 := @parameter0: java.io.PrintStream;

        specialinvoke r0.<org.xmlpull.v1.builder.XmlBuilderException: void printStackTrace()>();

        $r1 = r0.<org.xmlpull.v1.builder.xpath.jaxen.FunctionCallException: java.lang.Throwable nestedException>;

        if $r1 == null goto label1;

        virtualinvoke r2.<java.io.PrintStream: void println(java.lang.String)>("Root cause:");

        $r3 = r0.<org.xmlpull.v1.builder.xpath.jaxen.FunctionCallException: java.lang.Throwable nestedException>;

        virtualinvoke $r3.<java.lang.Throwable: void printStackTrace(java.io.PrintStream)>(r2);

     label1:
        return;
    }

    public void printStackTrace(java.io.PrintWriter)
    {
        java.lang.Throwable $r1, $r3;
        java.io.PrintWriter r2;
        org.xmlpull.v1.builder.xpath.jaxen.FunctionCallException r0;

        r0 := @this: org.xmlpull.v1.builder.xpath.jaxen.FunctionCallException;

        r2 := @parameter0: java.io.PrintWriter;

        specialinvoke r0.<org.xmlpull.v1.builder.XmlBuilderException: void printStackTrace()>();

        $r1 = r0.<org.xmlpull.v1.builder.xpath.jaxen.FunctionCallException: java.lang.Throwable nestedException>;

        if $r1 == null goto label1;

        virtualinvoke r2.<java.io.PrintWriter: void println(java.lang.String)>("Root cause:");

        $r3 = r0.<org.xmlpull.v1.builder.xpath.jaxen.FunctionCallException: java.lang.Throwable nestedException>;

        virtualinvoke $r3.<java.lang.Throwable: void printStackTrace(java.io.PrintWriter)>(r2);

     label1:
        return;
    }

    public void printStackTrace()
    {
        java.lang.Throwable $r1, $r3;
        java.io.PrintStream $r2;
        org.xmlpull.v1.builder.xpath.jaxen.FunctionCallException r0;

        r0 := @this: org.xmlpull.v1.builder.xpath.jaxen.FunctionCallException;

        specialinvoke r0.<org.xmlpull.v1.builder.XmlBuilderException: void printStackTrace()>();

        $r1 = r0.<org.xmlpull.v1.builder.xpath.jaxen.FunctionCallException: java.lang.Throwable nestedException>;

        if $r1 == null goto label1;

        $r2 = <java.lang.System: java.io.PrintStream out>;

        virtualinvoke $r2.<java.io.PrintStream: void println(java.lang.String)>("Root cause:");

        $r3 = r0.<org.xmlpull.v1.builder.xpath.jaxen.FunctionCallException: java.lang.Throwable nestedException>;

        virtualinvoke $r3.<java.lang.Throwable: void printStackTrace()>();

     label1:
        return;
    }

    public java.lang.Throwable fillInStackTrace()
    {
        java.lang.Throwable $r1, $r2, $r3, $r4;
        org.xmlpull.v1.builder.xpath.jaxen.FunctionCallException r0;

        r0 := @this: org.xmlpull.v1.builder.xpath.jaxen.FunctionCallException;

        $r1 = r0.<org.xmlpull.v1.builder.xpath.jaxen.FunctionCallException: java.lang.Throwable nestedException>;

        if $r1 != null goto label1;

        $r4 = specialinvoke r0.<java.lang.Throwable: java.lang.Throwable fillInStackTrace()>();

        return $r4;

     label1:
        $r2 = r0.<org.xmlpull.v1.builder.xpath.jaxen.FunctionCallException: java.lang.Throwable nestedException>;

        $r3 = virtualinvoke $r2.<java.lang.Throwable: java.lang.Throwable fillInStackTrace()>();

        return $r3;
    }

    public java.lang.Throwable getNestedException()
    {
        java.lang.Throwable $r1;
        org.xmlpull.v1.builder.xpath.jaxen.FunctionCallException r0;

        r0 := @this: org.xmlpull.v1.builder.xpath.jaxen.FunctionCallException;

        $r1 = r0.<org.xmlpull.v1.builder.xpath.jaxen.FunctionCallException: java.lang.Throwable nestedException>;

        return $r1;
    }
}
